tools/llvmbuildtobzl/llvmbuildtobzl.go

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	"bitbucket.org/creachadair/stringset"
+	"github.com/creachadair/ini"
+	"github.com/kythe/llvmbzlgen/path"
+	"github.com/kythe/llvmbzlgen/writer"
+)
+
+var (
+	camelPattern = regexp.MustCompile("(.)([A-Z][a-z]+)")
+	stringProps  = stringset.New("name", "parent", "library_name")
+	listProps    = stringset.New("dependencies", "required_libraries", "add_to_library_groups")
+)
+
+type iniFile map[string]iniSection
+type iniSection map[string][]string
+type propArgs map[string]interface{}
+
+func load(path string) (iniFile, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	f := make(iniFile)
+	return f, ini.Parse(file, ini.Handler{
+		Section: func(_ ini.Location, name string) error {
+			f[name] = make(iniSection)
+			return nil
+		},
+		KeyValue: func(loc ini.Location, key string, values []string) error {
+			f[loc.Section][key] = flatSplit(values)
+			return nil
+		},
+	})
+}
+
+func (f *iniFile) Subdirectories() []string {
```

Comment:
Since `iniFile` is already a map, there's no need to use a pointer for its methods—it's already implicitly a reference.